wso2-synapse-unit-test-plugin/src/main/java/org/wso2/synapse/unittest/UnitTestCasesMojo.java

```diff
@@ -111,55 +154,111 @@ private void testCaseRunner() throws IOException {
      */
-    private void isAssertResults(String response, String synapseTestCaseFile) throws IOException {
-        String expectedResponse = "{\"test-cases\":\"SUCCESS\"}";
-
-        if (!response.equals(expectedResponse)) {
-            throw new IOException("Test Cases are failed - " + response);
-        }
+    private void appendTestLogs() {
+        getLog().info("------------------------------------------------------------------------");
+        getLog().info("U N I T - T E S T S");
+        getLog().info("------------------------------------------------------------------------");
 
-        getLog().info("SynapseTestCaseFile " + synapseTestCaseFile + " passed successfully");
-        getLog().info(" ");
     }
 
     /**
-     * Append log of UNIT-TEST when testing started.
+     * Generate unit test report.
+     *
+     * @param summaryData summary data of the unit test received from synapse server
      */
-    private void appendTestLogs() {
+    private void generateUnitTestReport(Map<String, String> summaryData) throws IOException {
+        if (summaryData.isEmpty()) {
+            return;
+        }
+
         getLog().info("------------------------------------------------------------------------");
-        getLog().info("U N I T - T E S T S");
+        getLog().info("U N I T - T E S T  R E P O R T");
         getLog().info("------------------------------------------------------------------------");
 
+        getLog().info("Start Time: " + dateFormat.format(timeStarted));
+
+
+        Date timeStop = new Date();
+        long duration = timeStop.getTime() - timeStarted.getTime();
+        getLog().info("Test Run Duration: " + TimeUnit.MILLISECONDS.toSeconds(duration) + " seconds");
+        getLog().info("Test Summary: ");
+        getLog().info("");
+
+        List<Boolean> testFailedSuccessList = new ArrayList<>();
+        for (Map.Entry<String, String> summary : summaryData.entrySet()) {
+            String fileSeperatePattern = Pattern.quote(System.getProperty(Constants.FILE_SEPARATOR));
+            String[] filePathParams = summary.getKey().split(fileSeperatePattern);
+            String testName = filePathParams[filePathParams.length - 1];
+            getLog().info("Test Suite Name: " + testName);
+            getLog().info("==============================================");
+
+            //convert response from synapse engine as a json object
+            JsonObject summaryJson = new JsonParser().parse(summary.getValue()).getAsJsonObject();
+
+            //calculate pass and failure test counts
+            Map.Entry<String, String> testFailPassCounts = getPassFailureTestCaseCounts(summaryJson);
+            String passTestCaseCount = testFailPassCounts.getKey();
+            String failureTestCaseCount = testFailPassCounts.getValue();
+
+            getLog().info("Pass Test Cases: " + passTestCaseCount);
+            getLog().info("Failure Test Cases: " + failureTestCaseCount);
+            getLog().info("");
```

Comment:
please what is the ussage of this empty log

Reply:
To keep the empty line with [INFO] tag.